fileio/ft_read_event.m

```diff
@@ -1362,9 +1362,15 @@
     end
     
   case 'matlab'
-    % read the events from a normal MATLAB file
-    tmp   = load(filename, 'event');
-    event = tmp.event;
+    % read the event structure from a MATLAB file
+    % it should either contain an "event" structure, or a FieldTrip data structure according to FT_DATATYPE_RAW
+    w = whos(matfile(filename));
+    if any(strcmp({w.name}, 'event'))
+      event = loadvar(filename, 'event');
+    elseif any(strcmp({w.name}, 'data')) || length(w)==1
+      data = loadvar(filename, 'data');
+      event = ft_fetch_event(data);
```

Comment:
this can be further improved for the case that the data structure does not have events yet (in cfg.previous.previous ...). In that case ft_read_event could also do the onset/offset detection on trigger channels using the low-level read_event function.

i.e. ft_read_event calls private/read_trigger, which calls ft_read_header+ft_read_data to get the trigger channel, threshold, onset-detect, and construct the event structure on the fly.